### PR TITLE
Implement isempty for Network

### PR DESCRIPTION
**Pull request by BSnelling**
The definition I've implemented is that a `Network` is empty if all its `Records` fields are empty.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -459,4 +459,31 @@ using Test
         @test length(net.buses) == 2
         @test length(net.loads) == 1
     end
+
+    @testset "empty network" begin
+
+        empty_net = Network(
```
Comment (BSnelling) [inline test/runtests.jl:465]: Slightly long winded test set up.  Is there perhaps a tidier constructor for this?

File: src/types.jl [hunk 2]
```diff
@@ -2846,6 +2846,15 @@ for R in (
     @eval $R(sizehint::Integer=0) = $R(map(T -> sizehint!(T(), sizehint), fieldtypes($R))...)::$R
 end
 
+function Base.isempty(net::T) where {T <: Network}
+    for i in 1:fieldcount(T)
+        if typeof(getfield(net, i)) <: Records && !isempty(getfield(net, i))
```
Comment (nickrobinson251) [inline src/types.jl:2851]: what do you think of either
```suggestion
        x = getfield(net, i)
        if x isa Records && !isempty(x)
```
or
```suggestion
        if fieldtype(net, i) <: Records && !isempty(getfield(net, i))
```
?

File: test/runtests.jl (hunk 1 above)
Comment (nickrobinson251) [reply test/runtests.jl:465]: yeah... i think it's okay

I suppose an alternative would be to define `Network()` as a way to create an empty `Network`
and/or implement `empty!(x::Network)`

But i'm fine with this as it is

File: test/runtests.jl [hunk 3]
```diff
@@ -459,4 +459,31 @@ using Test
...
+            ImpedanceCorrections(),
+            MultiTerminalDCLines(),
+            MultiSectionLineGroups30(),
+            Zones(),
+            InterAreaTransfers(),
+            Owners(),
+            FACTSDevices30(),
+        )
+
+        @test isempty(empty_net)
```
Comment (nickrobinson251) [inline test/runtests.jl:487]: let's also test the non-empty case e.g. using one of the test files

**Review by nickrobinson251 — COMMENTED** [review]
thank you!

closes #68

File: src/types.jl (hunk 2 above)
Comment (BSnelling) [reply src/types.jl]: Using `fieldtype` sounds good, thanks!

File: test/runtests.jl (hunk 3 above)
Comment (BSnelling) [reply test/runtests.jl:489]: Added

File: test/runtests.jl [hunk 4]
```diff
@@ -459,4 +459,34 @@ using Test
...
     end
+
+    @testset "empty network" begin
+
+        empty_net = Network(
+            30,
+            CaseID(),
+            Buses30(),
+            Loads(),
+            FixedShunts(),
```
Comment (nickrobinson251) [inline test/runtests.jl:470]: technically a v30 file doesn't have `FixedShunts`, so for the sake of making tests an accurate example:
```suggestion
            nothing,
```

**Review by nickrobinson251 — APPROVED** [review]
Thanks again! 

One final comment, then LGTM and i'll merge :)
